s2a/src/main/java/io/grpc/s2a/handshaker/S2AProtocolNegotiatorFactory.java

```diff
@@ -227,7 +228,9 @@ protected void handlerAdded0(ChannelHandlerContext ctx) {
             @Override
             public void onSuccess(SslContext sslContext) {
               ChannelHandler handler =
-                  InternalProtocolNegotiators.tls(sslContext).newHandler(grpcHandler);
+                  InternalProtocolNegotiators.tls(
+                          sslContext, new FixedObjectPool<>(Executors.newFixedThreadPool(1)))
```

Comment:
This creates a new thread, and doesn't shut it down, for each handshake. If we want to do this temporarily (just as a stepping stone), that is fine, but we'll need to do something better here. (So you can worry about this some in a future PR.)

Since this is client-side, it seems fairly safe to just use an (unlimited) cached thread pool across the connections. That could be served by GrpcUtil.SHARED_CHANNEL_EXECUTOR.

Reply:
Thanks for the suggestion, it SGTM, I've changed it to use the Shared thread pool: https://github.com/grpc/grpc-java/pull/11539/commits/168d62f520f74d6f1e919a5e779d99de6becc502